From 42a76df77e1da2a01e27d8b3a08e373584f2ed98 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
Date: Mon, 4 Mar 2019 10:02:27 +0700
Subject: [PATCH 8/9] hmi: demos: encoded preview: Change videoconvert to
 compatible

RCar-Gen3 not support vspmfilter, change to vspfilter.
Also set size of display by vspfilter instead of waylandsink

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@rvc.renesas.com>
---
 RZ_GST-demo_encoded-review.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/RZ_GST-demo_encoded-review.sh b/RZ_GST-demo_encoded-review.sh
index d5dd477..76efac1 100755
--- a/RZ_GST-demo_encoded-review.sh
+++ b/RZ_GST-demo_encoded-review.sh
@@ -14,7 +14,7 @@ fi
 function replay()
 {
 	printf "\e[34mPlaying... $1\e[0m\n"
-	gst-launch-1.0 -e filesrc location=$1 ! h264parse ! omxh264dec ! waylandsink out-width=800 out-height=480 qos=false
+	gst-launch-1.0 -e filesrc location=$1 ! qtdemux ! queue ! h264parse ! omxh264dec ! vspfilter ! video/x-raw,width=800,height=480 ! waylandsink qos=false
 }
 
 function record()
@@ -24,11 +24,11 @@ function record()
 	v4l2src num-buffers=$((TIME*30)) device=$CAM_DEV ! \
 	video/x-raw,format=UYVY,width=720,height=480 ! \
 	tee name=t t. ! \
-	vspmfilter outbuf-alloc=true ! \
+	vspfilter ! \
 	video/x-raw,format=NV12,width=640,height=480 ! \
 	omxh264enc target-bitrate=10485760 ! \
 	filesink location=$1 t. ! \
-	vspmfilter ! video/x-raw,format=BGRA,width=800,height=480 ! \
+	vspfilter ! video/x-raw,format=BGRA,width=800,height=480 ! \
 	waylandsink qos=false
 }
 
-- 
2.7.4

